### ISSUE TITLE
Add CDI dependenci to the API module

### ISSUE THREAD

**Issue by xstefank** [eclipse/microprofile-health#215]
Since every HealthCheck must be a CDI bean would it make sense to add CDI dependency directly to the API module? Then to start with MP Health all you need is only one dependency (Health API).

**Comment by donbourne** [comment]
Is it possible today, without CDI, to have `/health/ready` and `/health/live` simply answer whether the server is up and app(s) is running without any explicit HealthChecks and without CDI?  If so, I'd be hesitant to add an explicit dependency on CDI for the Health API.

**Comment by xstefank** [comment]
@donbourne yes, you are right. However, the question then is whether users rather use default behavior (no checks executed, liveness and readiness are the same, so only HC is used as a normal ping) or they define their own custom checks.
